batch10/assignment2/tamkeen-zarah.md

```diff
@@ -0,0 +1,55 @@
+# Q1) Share screenshot of the above Branch Protection of main branch
+
+![Branch Protection of main branch](https://user-images.githubusercontent.com/62700476/194774394-a514f13d-35b8-42e2-90a0-0504059f3fad.png)
+
+
+# Q2) Explain Docker Containers vs VMs
+
+## Docker Containers
+
+- OS level virtualisation
+- light weight
+- Fast
+- Simply isolated processes 
+- Memory efficient
+
+## VMs
+
+- Hardware level virtualisation
+- Shared apps
+- Dedicated memory
+
+# Q3) Explain Docker Architecture
+
+- Docker is a tool used for containerization
+## Client
+● A CLI fro interaction with Daemon
+
+## Daemon
+● Main component running and communicating with OS
+● Build and store Images
+● It can create,ship and run containers
+
+## Registry
+● Sharing of images
+● DockerHub, private registries
+
+# Q4) Write command to create an nginx container in detached mode with name assignment-2 running on host port 9090 and container port 80 on a custom network named assignment-2
+
+sudo docker network create 9090:80
```

Comment:
this is not correct command, `9090:80` is to publish ports, not create network...
the correct command is `docker network create NETWORK_NAME`

Reply:
is it correct  now